src/lib/simulation/toa.ts

```diff
@@ -0,0 +1,1581 @@
+import { UserStats, XpGainSource } from '@prisma/client';
+import { bold } from 'discord.js';
+import {
+	calcPercentOfNum,
+	calcWhatPercent,
+	increaseNumByPercent,
+	objectEntries,
+	percentChance,
+	randArrItem,
+	randInt,
+	reduceNumByPercent,
+	roll,
+	round,
+	sumArr,
+	Time
+} from 'e';
+import { CommandResponse } from 'mahoji/dist/lib/structures/ICommand';
+import { Bank, LootTable } from 'oldschooljs';
+import SimpleTable from 'oldschooljs/dist/structures/SimpleTable';
+
+import { mahojiParseNumber, userStatsBankUpdate } from '../../mahoji/mahojiSettings';
+import { mentionCommand } from '../commandMention';
+import { Emoji } from '../constants';
+import { getSimilarItems } from '../data/similarItems';
+import { degradeItem } from '../degradeableItems';
+import { GearStats, UserFullGearSetup } from '../gear/types';
+import { trackLoot } from '../lootTrack';
+import { MUserClass } from '../MUser';
+import { setupParty } from '../party';
+import { getMinigameEntity, getMinigameScore } from '../settings/minigames';
+import { SkillsEnum } from '../skilling/types';
+import { constructGearSetup, Gear } from '../structures/Gear';
+import { ItemBank, MakePartyOptions, Skills } from '../types';
+import { TOAOptions } from '../types/minions';
+import {
+	assert,
+	channelIsSendable,
+	clamp,
+	formatDuration,
+	formatSkillRequirements,
+	itemNameFromID,
+	randomVariation
+} from '../util';
+import addSubTaskToActivityTask from '../util/addSubTaskToActivityTask';
+import getOSItem from '../util/getOSItem';
+import itemID from '../util/itemID';
+import resolveItems from '../util/resolveItems';
+import { exponentialPercentScale, scaleNumber } from '../util/smallUtils';
+import { updateBankSetting } from '../util/updateBankSetting';
+import { TeamLoot } from './TeamLoot';
+
+const teamSizeScale: Record<number, number> = {
+	1: 1.0,
+	2: 1.9,
+	3: 2.8,
+	4: 3.4,
+	5: 4.0,
+	6: 4.6,
+	7: 5.2,
+	8: 5.8
+};
+const averageWTPLevel = { 0: 0, 1: 0, 2: 0, 3: 0 };
+
+function scaleHP(base: number, raidLevel: number, teamSize: keyof typeof teamSizeScale, pathLevel: number): number {
+	let pathMultiplier = 1.0;
+	if (pathLevel > 0) {
+		pathMultiplier = 1.03 + pathLevel * 0.05;
+	}
+	return Math.floor(base * (1 + raidLevel / 250) * teamSizeScale[teamSize] * pathMultiplier);
+}
+
+function estimatePoints(raidLevel: number, teamSize: number) {
```

Comment:
is this the Tombs of Amascut loot calculator on wiki? 

https://oldschool.runescape.wiki/w/Calculator:Tombs_of_Amascut_loot

since I heard some people saying the points wasn't scaling accordingly